### ISSUE TITLE
Log via Systemd is not possible anymore

### ISSUE THREAD

**Issue by razer4908** [oznu/homebridge-config-ui-x#106]
Protocol Output:

> Using systemd to tail logs
> CMD: sudo -n journalctl -o cat -n 500 -f -u homebridge
> Failed to get data: The resolved adress can not be assigned 

Is there an easy solution for this ? 

Greetings

**Comment by oznu** [comment]
Hi @razer4908,

What happens if you run the command from the terminal directly?

```
sudo -n journalctl -o cat -n 500 -f -u homebridge
```

**Comment by razer4908** [comment]
This will give me the exact same error

**Comment by oznu** [comment]
I updated my system and can't replicate this. If it's not working directly at the terminal either then it's probably not a problem with this plugin.

 Is journalctl how you normally view logs for homebridge?

**Comment by razer4908** [comment]
Yes, normally this works very well. Since Update 3.6.6 i think an error occured when i try to view the logs. In this time no other changes have been configured.

**Comment by URBANsUNITED** [comment]
Maybe its linked to my original problem, which is back again... And I've installed everything from the scratch, problem is back . Very strange...
#87 

Journalctl, update, service restart etc. is working fine from terminal but doesn't from web gui/plugin.

**Comment by oznu** [comment]
Make sure the owner of the homebridge data directory is the same as the user running homebridge. The log shell "spawns" with this as the current working directory.

**Comment by razer4908** [comment]
user is the same in my config. i only have the "root" user (i know ... not best practice)  I worked some days before. Very strange

**Comment by URBANsUNITED** [comment]
Here is my overview of files and permissions. Maybe there is an error inside...

**/etc/default/homebridge**

Defaults / Configuration options for homebridge
The following settings tells homebridge where to find the config.json file and where to$
HOMEBRIDGE_OPTS=-U /var/lib/homebridge -I
If you uncomment the following line, homebridge will log more
You can display this via systemd's journalctl: journalctl -f -u homebridge
DEBUG=*

**/etc/systemd/system/homebridge.service**

[Unit]
Description=Node.js HomeKit Server
After=syslog.target network-online.target

[Service]
Type=simple
User=homebridge
EnvironmentFile=/etc/default/homebridge
Adapt this to your specific setup (could be /usr/bin/homebridge)
See comments below for more information
ExecStart=/usr/bin/homebridge $HOMEBRIDGE_OPTS
Restart=on-failure
RestartSec=10
KillMode=process
CapabilityBoundingSet=CAP_NET_RAW
AmbientCapabilities=CAP_NET_RAW

[Install]
WantedBy=multi-user.target

**id homebridge**
uid=1001(homebridge) gid=1001 Gruppen=1001,0(root),27(sudo),997(gpio)

**ls /var/lib/homebridge -l**

insgesamt 560
drwxrwxrwx 2 homebridge homebridge   4096 Mai 20 23:24 accessories
-rwxrwxrwx 1 homebridge homebridge    359 Mai  9 18:32 auth.json
-rw-r--r-- 1 homebridge       1001   7687 Jun  1 19:39 config.json
-rwxrwxrwx 1 homebridge homebridge   8815 Mai  1 07:53 config.json.1525155968507
-rwxrwxrwx 1 homebridge homebridge   8814 Mai  1 08:26 config.json.1525541705562
-rwxrwxrwx 1 homebridge homebridge   8888 Mai  5 19:35 config.json.1525543343162
-rwxrwxrwx 1 homebridge homebridge   8873 Mai  5 20:02 config.json.1525543385398
-rwxrwxrwx 1 homebridge homebridge   8873 Mai  5 20:03 config.json.1525881505924
-rwxrwxrwx 1 homebridge homebridge   8956 Mai  9 17:58 config.json.1525883573478
-rwxrwxrwx 1 homebridge homebridge   8956 Mai  9 18:32 config.json.1525892306350
-rwxrwxrwx 1 homebridge homebridge   8947 Mai  9 20:58 config.json.1525892500049
-rwxrwxrwx 1 homebridge homebridge   7078 Mai 20 23:32 config.json.1526853476761
-rw-r--r-- 1 homebridge homebridge   7135 Mai 20 23:57 config.json.1526853560306
-rw-r--r-- 1 homebridge homebridge   7137 Mai 20 23:59 config.json.1526853632923
-rw-r--r-- 1 homebridge homebridge   7139 Mai 21 00:00 config.json.1527435984128
-rw-r--r-- 1 homebridge homebridge   7864 Mai 27 17:46 config.json.1527874795115
-rwxrwxrwx 1 homebridge homebridge 245938 Mai 10 09:13 Homebridge_Now_persist.json
drwxrwxrwx 2 homebridge homebridge   4096 Mai 20 23:24 persist
-rw-r--r-- 1 homebridge homebridge 148806 Jun  6 22:33 raspberrypi_Now_persist.json
-rwxrwxrwx 1 homebridge homebridge   9470 Mär  9 11:51 Raspi3_Now_persist.json
pi@raspberrypi:~ $ 

**ls /etc/default/ -l**

insgesamt 92
-rw-r--r-- 1 root root  556 Mär 30 17:07 apache-htcacheclean
-rw-r--r-- 1 root root  219 Jan 23  2017 avahi-daemon
-rw-r--r-- 1 root root  845 Okt 24  2016 bluetooth
-rw-r--r-- 1 root root  222 Okt 25  2016 bsdmainutils
-rw-r--r-- 1 root root  276 Apr 18 02:27 console-setup
-rw-r--r-- 1 root root  549 Aug 23  2014 crda
-rw-r--r-- 1 root root  955 Okt  7  2017 cron
-rw-r--r-- 1 root root  297 Mär  2 09:59 dbus
-rw-r--r-- 1 root root 1544 Okt  2  2017 dnsmasq
-rw-r--r-- 1 root root   84 Mär  5  2012 fake-hwclock
-rw-r--r-- 1 root root  378 Mai 28 06:50 homebridge
...

THANKS
URBANsUNITED

**Comment by URBANsUNITED** [comment]
FINALLY! Got it working...

The problems is this:
This passage is not working for me:

> Password-less sudo required
> For sudo mode to work password-less sudo is required. You can enable password-less sudo by 
> adding this entry to the bottom of your /etc/sudoers file (use visudo to edit the file!):
> 
> homebridge    ALL=(ALL) NOPASSWD: ALL

I've fixed it by adding this

homebridge ALL=(root) SETENV:NOPASSWD: /usr/bin/npm, /bin/systemctl restart homebridge, /bin/journalctl, /usr/bin/node

to the /etc/sudoers.d/homebridge (which isn't existing before) by sudo nano /etc/sudoers.d/homebridge
Maybe for others this not working. check where files are located:

which homebridge
/usr/bin/homebridge

if /usr/local/bin/homebridge
Add local to /usr/bin = /usr/local/bin

Restarted the server and bams all is working as it should!

Problem fixed!

Thanks for your patience

URBANsUNITED

**Comment by oznu** [comment]
Thanks @URBANsUNITED, I'll do some testing and might update the README.

@razer4908 - I think your issue is very different, since your unable to even run the command manually at the command line there is something else wrong outside the control of this plugin.
